prometheus/src/lib.rs

```diff
@@ -1,21 +1,27 @@
 mod bank_metrics;
+pub mod banks_with_commitments;
 mod cluster_metrics;
 mod utils;
 
+use banks_with_commitments::BanksWithCommitments;
 use solana_gossip::cluster_info::ClusterInfo;
-use solana_runtime::bank_forks::BankForks;
-use std::sync::{Arc, RwLock};
+use std::sync::Arc;
 
 pub struct Lamports(pub u64);
 
 pub fn render_prometheus(
-    bank_forks: &Arc<RwLock<BankForks>>,
+    banks_with_commitments: BanksWithCommitments,
     cluster_info: &Arc<ClusterInfo>,
 ) -> Vec<u8> {
-    let current_bank = bank_forks.read().unwrap().working_bank();
+    // There are 3 levels of commitment for a bank:
+    // - finalized: most recent block *confirmed* by supermajority of the
```

Comment:
Validators don’t “confirm” blocks, they only vote on blocks.

Finalized means that there are 32 descendants of the block that a supermajority of the cluster have voted on. (I might be off by one here.) I now also understand what all this “rooted” and “root slot” business is: they call the highest slot that is finalized the “root”. I suppose because it’s the root of the bank forks, other blocks on top could in theory be replaced by some other fork, but for the “root” the timeout is so long that we don’t consider it revertable.

Reply:
Well.. This is the copy-paste from their docs 😅